### PR TITLE
Init the project structure

### PR DESCRIPTION
**Pull request by kalonice**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,2 @@
...
-CMakeFiles
-CMakeScripts
-Testing
-Makefile
-cmake_install.cmake
-install_manifest.txt
-compile_commands.json
-CTestTestfile.cmake
+.idea
+build
```
Comment (alexey-sidnev) [inline .gitignore]: I believe it's not good idea ignore `build`. Better exclude files by extension: 
 - *.dll
 - *.exe
 - *.o
 - *.so

Remove `build` or add exceptions which we really must exclude.

File: 3rdparty/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+file(GLOB_RECURSE GTEST_CPPS "gtest/googletest/src/*.cc")
+file(GLOB_RECURSE GTEST_HPPS "gtest/googletest/include/*.h")
+
+include_directories(./gtest/googletest/include)
+include_directories(./gtest/googletest/)
```
Comment (alexey-sidnev) [inline 3rdparty/CMakeLists.txt:5]: Have you tried to build it in Windows? I believe it's not so easy. Take a look on my repository.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+language: cpp python
+
+python: '2.7'
+
+sudo: false
+
+compiler:
+  - gcc
```
Comment (alexey-sidnev) [inline .travis.yml:8]: Which version of gcc will we use? We need C++11 and C++14.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.2)
+
+project(training)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (alexey-sidnev) [inline CMakeLists.txt]: Looks like CMAKE is really stupid - you repeat requirements twice)

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
```
Comment (alexey-sidnev) [inline CMakeLists.txt]: You should add `add_compile_options(-Wall -Wpedantic -Wextra -Werror)`. [Link](https://github.com/alexey-sidnev/infrastructure/blob/master/CMakeLists.txt).

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	if [ ! -d "$(BUILD_DIR)" ]; then mkdir $(BUILD_DIR); fi
+	cd $(BUILD_DIR)
+
+ut: deps
+	set -e
+	cd $(BUILD_DIR) && cmake ../ -DBUILD_TESTS=ON && make -j ${PROCESSORS} ${TEST_EXECUTABLE_NAME}
+	$(BUILD_DIR)/bin/${TEST_EXECUTABLE_NAME}
+
+app: deps
+	set -e
```
Comment (alexey-sidnev) [inline Makefile]: We don't need Makefile at all.

**Review by alexey-sidnev — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (zingorn) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 5 above)
Comment (zingorn) [reply CMakeLists.txt]: Done

Comment (zingorn) [reply CMakeLists.txt]: Done

File: .travis.yml (hunk 3 above)
Comment (zingorn) [reply .travis.yml:8]: Done

File: 3rdparty/CMakeLists.txt (hunk 2 above)
Comment (alexey-sidnev) [reply 3rdparty/CMakeLists.txt:5]: Ok, will fix next time.

**Review by alexey-sidnev — APPROVED** [review]

**Pull request merged**